vm/src/types/slot.rs

```diff
@@ -163,31 +163,37 @@ fn as_mapping_wrapper(zelf: &PyObject, _vm: &VirtualMachine) -> PyMappingMethods
         };
     }
     PyMappingMethods {
-        length: then_some_closure!(zelf.has_class_attr("__len__"), |zelf, vm| {
-            vm.call_special_method(zelf, "__len__", ()).map(|obj| {
-                obj.payload_if_subclass::<PyInt>(vm)
-                    .map(|length_obj| {
-                        length_obj.as_bigint().to_usize().ok_or_else(|| {
-                            vm.new_value_error("__len__() should return >= 0".to_owned())
+        length: then_some_closure!(zelf.has_class_attr("__len__"), |mapping, vm| {
+            vm.call_special_method(mapping.obj.to_owned(), "__len__", ())
+                .map(|obj| {
+                    obj.payload_if_subclass::<PyInt>(vm)
+                        .map(|length_obj| {
+                            length_obj.as_bigint().to_usize().ok_or_else(|| {
+                                vm.new_value_error("__len__() should return >= 0".to_owned())
+                            })
                         })
-                    })
-                    .unwrap()
-            })?
+                        .unwrap()
+                })?
+        }),
+        subscript: then_some_closure!(zelf.has_class_attr("__getitem__"), |mapping, needle, vm| {
+            vm.call_special_method(mapping.obj.to_owned(), "__getitem__", (needle.to_owned(),))
```

Comment:
Although it is not directly relevant with this PR, as we update all the mapping methods it became incompatibility between cpython 3.10.

```python
>>> class TestDict(dict):
...     def __setitem__(self, key, value):
...             pass

>>> test1 = dict([[1,2],[3,4]])
>>> test2 = TestDict([[1,2],[3,4]])
```
##### RustPython
```python
>>>>> id(test1)
4821477152
>>>>> id(test2)
4318054400
```
##### cpython
```python
>>> id(test1)
4308420304
>>> id(test2)
4308420304
```
As we update `__setitem__` only, `__getitem__` must be same with base class.

I don't know what this difference makes problem yet. but just for mind up 😊